### PR TITLE
Feature/hash order ids

### LINKED ISSUE #3: Worker store-config orderId sequence generation should be hashed
Introducing hasing to the order ID generation (/workers/store-config) such that
- Order ID increments are not sequential
- Hash does not need to be reversible
- Hash should not conflict 99.9999999999% of time
- Do not store the hash value in DO, instead store the unhashed
- Hash length should be the same as the input length (or very similar but still same every time), max 10
- Preseve the prefix HT/HTT (not included in input length)

### PR DESCRIPTION
**Pull request by KYLChiu**

### CODE REVIEW

File: workers/store-config/src/objects/OrderIdObject.ts [hunk 1]
```diff
@@ -7,53 +7,57 @@ export class OrderIdObject {
...
-        ) {
-            nextSeq = getRndInteger(2, 10);
-        } else {
-            nextSeq = parseInt(storedValue.slice(HT_PREFIX.length + 6)) + 1;
-        }
-        nextSeq = nextSeq.toString().padStart(3, '0');
+
+        // Current date in YYYMMDDHHmmSSms format
+        const date = new Date().toISOString();
+        console.log(date);
```
Comment (KYLChiu) [inline workers/store-config/src/objects/OrderIdObject.ts:40]: Will remove, my bad

File: workers/store-config/src/objects/OrderIdObject.ts [hunk 2]
```diff
@@ -7,53 +7,54 @@ export class OrderIdObject {
...
-            storedValue.slice(HT_PREFIX.length, HT_PREFIX.length + 6) !== date
-        ) {
-            nextSeq = getRndInteger(2, 10);
-        } else {
-            nextSeq = parseInt(storedValue.slice(HT_PREFIX.length + 6)) + 1;
-        }
-        nextSeq = nextSeq.toString().padStart(3, '0');
+
+        // Atomically generate and put the unique identifier
+        const nextSeq = await this.sequence();
```
Comment (KYLChiu) [inline workers/store-config/src/objects/OrderIdObject.ts:45]: [On second thought] Perhaps we should couple the padding length on line 43 with/as the sequence reset length - i.e. the padding length is passed to sequence(padding: int) and the reset is set at 1e(padding) -  let me know what you think

Comment (richardhunghhw) [reply workers/store-config/src/objects/OrderIdObject.ts:45]: Yes, either this or define the reset length at the class level. As long as its one place where 1e3 is defined

File: workers/store-config/src/objects/OrderIdObject.ts [hunk 3]
```diff
@@ -1,59 +1,68 @@
...
+        // Make value.
+        const result = new Date().toISOString() + nextSeqPadded;
+
+        // Obfusciate with hash, chance of collision is 16^hashLength.
+        const hashedResult =
+            HT_PREFIX +
+            (await this.hash(result)).slice(0, OrderIdObject.hashLength);
+
+        // Return response.
+        console.debug(`Generated Order ID: ${result}, DO value: ${nextSeq}.`);
```
Comment (richardhunghhw) [inline workers/store-config/src/objects/OrderIdObject.ts]: This log line is not printing the actual generated orderId result

Comment (KYLChiu) [reply workers/store-config/src/objects/OrderIdObject.ts]: Fixed!

**Pull request merged**